demos/common/cpp/pipelines/src/async_pipeline.cpp

```diff
@@ -65,15 +68,16 @@ int64_t AsyncPipeline::submitData(const InputData& inputData, const std::shared_
     auto frameID = inputFrameId;
 
     auto request = requestsPool->getIdleRequest();
-    if (!request)
+    if (!request) {
         return -1;
+    }
 
     auto startTime = std::chrono::steady_clock::now();
-    auto internalModelData = model->preprocess(inputData, request);
+    auto internalModelData = model->preprocess(inputData, *request);
     preprocessMetrics.update(startTime);
 
-    request->SetCompletionCallback(
-        [this, frameID, request, internalModelData, metaData, startTime]() {
+    request->set_callback(
+        [this, frameID, request, internalModelData, metaData, startTime](const std::exception_ptr& e) {
```

Comment:
```suggestion
        [this, frameID, request, internalModelData, metaData, startTime](std::exception_ptr e) {
```
That's the signature of expected callback func
Same for `~RequestsPool()`

Reply:
Ok, then it should be changed in other demos, where I already used  `const std::exception_ptr& e`. I'll do it later in another PR.
BTW, maybe you have explanation, why if we just set `{} `to callback some demos are stucking in the middle of run? https://github.com/openvinotoolkit/open_model_zoo/pull/3165#pullrequestreview-879206222

Reply:
Done.

Reply:
> BTW, maybe you have explanation, why if we just set `{} `to callback some demos are stucking in the middle of run? https://github.com/openvinotoolkit/open_model_zoo/pull/3165#pullrequestreview-879206222

You probably hit undefined behavior. It's a luck it works with unmatching signatures